apps/api/src/app/bridge/usecases/sync/sync.usecase.ts

```diff
@@ -111,21 +115,40 @@ export class Sync {
     createdWorkflows: NotificationTemplateEntity[]
   ): Promise<void> {
     const persistedWorkflowIdsInBridge = createdWorkflows.map((i) => i._id);
-
     const workflowsToDelete = await this.findAllWorkflowsWithOtherIds(command, persistedWorkflowIdsInBridge);
 
-    await Promise.all(
-      workflowsToDelete?.map((workflow) => {
-        return this.deleteWorkflow.execute(
-          DeleteWorkflowCommand.create({
-            environmentId: command.environmentId,
-            organizationId: command.organizationId,
-            userId: command.userId,
-            workflowId: workflow._id,
-          })
-        );
+    const deleteWorkflowFromStoragePromises = workflowsToDelete.map((workflow) =>
+      this.deleteWorkflow.execute(
+        DeleteWorkflowCommand.create({
+          environmentId: command.environmentId,
+          organizationId: command.organizationId,
+          userId: command.userId,
+          workflowId: workflow._id,
+        })
+      )
+    );
+    const invalidateCachesByIdentifiersPromises = workflowsToDelete.map((workflow) =>
```

Comment:
This invalidation should be added in the [delete-workflow usecase](https://github.com/novuhq/novu/pull/6887/files#diff-b97ae3759b8dec4d0c5057b0e10ca4798437b9c7896dacf049bf8da4de7f8000R130). The cache clearance should applies every time a workflow is deleted, not only via sync.

Reply:
Moreover, the delete-workflow use case needs to delete from the preferences collection all entries of type `WORKFLOW_RESOURCE` by _environmentId and workflow internal id. This is irrelevant to this bug, but as I reviewed the code, it's an omission we can easily tackle.

Reply:
> This invalidation should be added in the delete-workflow usecase. The cache clearance should applies every time a workflow is deleted, not only via sync.

First of all please note the last part i mentioned in this comment.
https://github.com/novuhq/novu/pull/6887#discussion_r1833005669

We can't invalidate these IDs because, in the delete workflow use case, we don't know which specific IDs need to be deleted. We're missing the current list of workflows that should be synced with the full snapshot list. Even if we had this data, we would still invalidate unnecessary records since we already know the exact IDs that need invalidation.

Reply:
> Moreover, the delete-workflow use case needs to delete from the preferences collection all entries of type WORKFLOW_RESOURCE by _environmentId and workflow internal id. This is irrelevant to this bug, but as I reviewed the code, it's an omission we can easily tackle.

Agree i will take a look at it.

Reply:
Done.
I deleted the old bridge.deleteWorkflow so we could reuse one reusable delete workflow from app-gen.